src/differentiation.jl

```diff
@@ -0,0 +1,139 @@
+
+"""
+    AbstractDiffBackend
+
+An abstract type for diff backends. See [`FiniteDifferencesBackend`](@ref) for
+an example.
+"""
+abstract type AbstractDiffBackend end
+
+struct NoneDiffBackend <: AbstractDiffBackend end
+
+
+"""
+    _derivative(f, t[, backend::AbstractDiffBackend])
+
+Compute the derivative of a callable `f` at time `t` computed using the given `backend`,
```

Comment:
We could add the formulae here, i.e. for a function $f\colon I \to \mathcal M$, where $I$ is some interval containing $t$, this computes $Df(t)$ and store that in `X`? And I think it is not (yet) the covariant one since this is the file on differentiation, i.e. in the embedding?

Reply:
Yes, this is actually just a unified way to call plain Euclidean differentiation methods (AD, finite differences), not Riemannian AD. I will add a comment about it to the documentation.